### DISCUSSION TITLE
WRF Hydro Using Custom Weather Model for Forcing Data

### DISCUSSION THREAD

**Discussion by zhong-industries** [NCAR/wrf_hydro_nwm_public#771] [General]
Hello WRF Hydro team,

I was looking at using a different weather model (none of the supported ones listed on your website: https://ral.ucar.edu/projects/wrf_hydro/pre-processing-tools) to use for forcing data into the WRF Hydro model. From the technical documentation, the following weather variables are required to run it:

| Variable name    | Description Units |
| -------- | ------- |
| SWDOWN Incoming shortwave radiation | W/m2 |
| LWDOWN Incoming longwave radiation | W/m2 |
| Q2D Specific humidity | kg/kg |
| T2D Air temperature | K |
| PSFC Surface pressure | Pa |
| U2D Near surface wind in the u-component | m/s |
| V2D Near surface wind in the v-component | m/s |
| RAINRATE Precipitation rate | mm/s or kg/m2 |

I have a couple questions regarding this:

1) At what height "level" do we need each variable. For some like Surface Pressure, Precipitation Rate, and the Shortwave / Longwave Radiation it makes sense to be getting the values for those at "surface level" from our model. However, for the others - Air temperature, Specific Humidity, Near Surface Wind, it's unclear to me what "levels" we should be using as the forcing data. 2 meters above ground level? 10 meters above ground level? 100 meters? A combination of those?

2) What kind of statistical processing / range should we try to get for the forcing data? I.e. when feeding in something like Air Temperature as forcing data, should we be getting Air Temperature at a single forecast point in time (instantaneous), or as a forecast average over a range in time (average)? Let's say we're running a forecast for 3 hours, is it okay to just get the following data (Temp @1 hr, Temp @2hr, Temp @3hr) or would we want something like (Temp Avg from [0-1]Hr, Temp Avg from [1-2]Hr, Temp Avg from [2-3]Hr)

**Comment by aubreyd** [comment]
Hi Alfred:
You might check the [Noah-MP technical description](https://opensky.ucar.edu/islandora/object/technotes:599) in case there are better answers there, but here is my understanding: 

1. There is a ZLVL setting in namelist.hrldas where the user can specify the height of the met inputs. My understanding is that the only variables this really impacts are the wind components, which get attenuated in the Noah-MP code based on measurement height.
2. For the most control you would probably want to setup your met inputs to match your LSM integration timestep. So if you want to integrate the LSM every 3 hours, you would want to feed it the (for example) the precipitation rate that most represents that 3-hour period. Whatever rate (or state) you feed the model will be used over its full integration timestep.

Hope that helps!
